src/test/regress/expected/multi_alter_table_add_constraints.out

```diff
@@ -719,19 +719,4 @@ SET LOCAL application_name to 'citus_internal gpid=10000000001';
 SET citus.enable_ddl_propagation TO OFF;
 -- alter table triggers SELECT, and auto_explain catches that
 ALTER TABLE target_table ADD CONSTRAINT fkey_167 FOREIGN KEY (col_1) REFERENCES test_ref_table(key) ON DELETE CASCADE;
-LOG:  duration: xxxx ms  plan:
-{
-  "Query Text": "SELECT fk.\"col_1\" FROM ONLY \"public\".\"target_table\" fk LEFT OUTER JOIN ONLY \"public\".\"test_ref_table\" pk ON ( pk.\"key\" OPERATOR(pg_catalog.=) fk.\"col_1\") WHERE pk.\"key\" IS NULL AND (fk.\"col_1\" IS NOT NULL)",
```

Comment:
Since we're not planning these queries anymore autoexplain doesn't create a plan for them either. Seems harmless to me, and probably we can remove the workaround introduced by @emelsimsek in #6470. @emelsimsek what are your thoughts on this?

Reply:
Agreed. The issue was due to the foreign key constraint check not being skipped in the planning phase. We should be able to remove the workaround. But let me test it first.

Reply:
Verified that the change in https://github.com/citusdata/citus/pull/6470 could be removed. It makes sense to remove it in this PR imo. I would keep the test case tweaking the comment. 
```
diff --git a/src/backend/distributed/planner/multi_explain.c b/src/backend/distributed/planner/multi_explain.c
index d7766a913..3d0fe617e 100644
--- a/src/backend/distributed/planner/multi_explain.c
+++ b/src/backend/distributed/planner/multi_explain.c
@@ -199,20 +199,6 @@ CitusExplainScan(CustomScanState *node, List *ancestors, struct ExplainState *es
                return;
        }

-       /*
-        * ALTER TABLE statements are not explained by postgres. However ALTER TABLE statements
-        * may trigger SELECT statements causing explain hook to run. This situation causes a crash in a worker.
-        * Therefore we will detect if we are explaining a triggered query when we are processing
-        * an ALTER TABLE statement and stop explain in this situation.
-        */
-       if (AlterTableInProgress())
-       {
-               ExplainPropertyText("Citus Explain Scan",
-                                                       "Explain for triggered constraint validation queries during ALTER TABLE commands are not supported by Citus",
-                                                       es);
-               return;
-       }
-
```

```
diff --git a/src/test/regress/expected/multi_alter_table_add_constraints.out b/src/test/regress/expected/multi_alter_table_add_constraints.out
index 5bd9499de..e1ec1da68 100644
--- a/src/test/regress/expected/multi_alter_table_add_constraints.out
+++ b/src/test/regress/expected/multi_alter_table_add_constraints.out
@@ -717,6 +717,6 @@ BEGIN;
 -- simulate being a worker session/backend
 SET LOCAL application_name to 'citus_internal gpid=10000000001';
 SET citus.enable_ddl_propagation TO OFF;
--- alter table triggers SELECT, and auto_explain catches that
+-- alter table should not trigger a SELECT query since skip_validation is turned on for distributed tables
 ALTER TABLE target_table ADD CONSTRAINT fkey_167 FOREIGN KEY (col_1) REFERENCES test_ref_table(key) ON DELETE CASCADE;
 END;
diff --git a/src/test/regress/sql/multi_alter_table_add_constraints.sql b/src/test/regress/sql/multi_alter_table_add_constraints.sql
index ccfc18fb9..ed72345b1 100644
--- a/src/test/regress/sql/multi_alter_table_add_constraints.sql
+++ b/src/test/regress/sql/multi_alter_table_add_constraints.sql
@@ -595,6 +595,6 @@ BEGIN;
 -- simulate being a worker session/backend
 SET LOCAL application_name to 'citus_internal gpid=10000000001';
 SET citus.enable_ddl_propagation TO OFF;
--- alter table triggers SELECT, and auto_explain catches that
+-- alter table should not trigger a SELECT query since skip_validation is turned on for distributed tables
 ALTER TABLE target_table ADD CONSTRAINT fkey_167 FOREIGN KEY (col_1) REFERENCES test_ref_table(key) ON DELETE CASCADE;
 END;
```

Reply:
>  It makes sense to revert it in this PR imo.

I'll revert this in a separate PR to make backporting this one easy. Since #6470 is a new change and would thus cause merge conflicts when cherry-picking to old release branches.